### PR TITLE
V3: OO continued

### PR DESCRIPTION
**Pull request by sougou**
@alainjobart 
generator kind of became very small, but I'm keeping it separate. I think it can accumulate some common wire-up code between various builders. It's renamed to jointab now.
I think next part will be separating out the engine. We'll see how that goes.

---
This change is [Reviewable](https://reviewable.io/reviews/youtube/vitess/1600)

### CODE REVIEW

File: go/vt/vtgate/planbuilder/jointab.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// ListVarName is the bind var name used for plans
+// that require VTGate to compute custom list values,
+// like for IN clauses.
+const ListVarName = "__vals"
+
+// jointab is used for managing join dependencies.
```
Comment (alainjobart) [inline go/vt/vtgate/planbuilder/jointab.go]: Can you add a comment on what it actually does? jointab name is kinda obscure...

**Comment by alainjobart** [comment]
LGTM with one comment.

File: go/vt/vtgate/planbuilder/jointab.go (hunk 1 above)
Comment (sougou) [inline go/vt/vtgate/planbuilder/jointab.go]: How about "jointab manages procurement and naming of join variables across primitives"?

Comment (alainjobart) [inline go/vt/vtgate/planbuilder/jointab.go]: Sure.

**Pull request merged**